third_party/winuwp_h264/H264Decoder/H264Decoder.cc

```diff
@@ -302,13 +304,19 @@ HRESULT WinUWPH264DecoderImpl::FlushFrames(uint32_t rtp_timestamp,
       }
 
       // Update members to avoid querying unnecessarily
-      width_.emplace(width);
-      height_.emplace(height);
+      output_width_.emplace(output_width);
+      output_height_.emplace(output_height);
     }
 
+    // Create an output buffer that has the same size as the input frame, even
+    // if the decoder produced a decoded frame of different size due to e.g.
+    // padding. This prevents returning some uninitialized data.
+    RTC_CHECK(input_width_.has_value());
```

Comment:
Previously, we were getting the input dimensions from the decoder itself in the event that they weren't provided by WebRTC, which happens in the case that the received packet header doesn't contain frame dimensions. In our case, frame dimensions were absent from the packet headers when using the Multiplex WebRTC protocol to send H264 frames, which we needed to do in order to support extra data on each frame for a customer.

Is there a way to read the non-padded dimensions of the decoded frame from Media Foundation? In theory, I'd think MF should know that information, even if the result will be padded. Perhaps we could otherwise copy the `output_*` dimensions in the event that `input_*` dimensions aren't set, rather than failing. That would be important to not break clients that don't provide input dimensions.

Reply:
I never considered WebRTC could not know the input frame size. I am surprised this can even work in any context; how is any processing supposed to work if given only a pointer to a buffer without the frame size? Or is it that for H.264 specifically that information is available inside the frame itself? Because for _e.g._ a raw I420 frame it is not, so unless WebRTC gives it to you, you cannot do anything with the buffer.

I didn't find any documentation on padding, just observed it on specific resolutions like 640x36**0** getting out as 640x36**8**, so assumed there was a 16-pixel padding.

Reply:
Wow that's disappointing regarding the absence of docs on the padding behavior. Any chance you have the bandwidth to report that as an issue to the product team?

For H.264, the encoded frame's data actually has enough information for a decoder to deduce the frame size.

Reply:
I'll try once we are done with microsoft/MixedReality-WebRTC#74, which is high priority, to avoid any confusion.